### PR TITLE
add unattended and sudoers options to blueprint

### PR DESCRIPTION
**Pull request by loadtheaccumulator**
# Description
Add unattended and sudoers options to the blueprint for ISO request

FIXES: THEEDGE-3922

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

File: pkg/clients/imagebuilder/client.go [hunk 1]
```diff
@@ -341,7 +348,13 @@ func (c *Client) ComposeInstaller(image *models.Image) (*models.Image, error) {
 	}
 	users := make([]User, 0)
 	if feature.PassUserToImageBuilder.IsEnabled() && image.Installer != nil && image.Installer.Username != "" && image.Installer.SSHKey != "" {
-		users = append(users, User{Name: image.Installer.Username, SSHKey: image.Installer.SSHKey})
+		user := &User{
```
Comment (ezr-ondrej) [inline pkg/clients/imagebuilder/client.go:351]: We define it as pointer here, but then dereferencing it... I like the inline style actually, I'm all for keeping it :)
```
users = append(users, User{
                        Name:   image.Installer.Username,
			SSHKey: image.Installer.SSHKey,
			Groups: []string{image.Installer.Username, "wheel"},
		}
```

If we need the extra var lets just have the it as value, not a pointer :)

```
user := User{ .... }
users = append(users, user)
```
```

File: pkg/clients/imagebuilder/client.go [hunk 2]
```diff
@@ -367,6 +380,16 @@ func (c *Client) ComposeInstaller(image *models.Image) (*models.Image, error) {
...
 	}
+
+	customInstaller := &CustomInstaller{
+		Unattended:   true,
+		SudoNoPasswd: []string{"@wheel"},
+	}
+
+	if feature.BlueprintCustomInstaller.IsEnabled() {
+		req.Customizations.Installer = customInstaller
+	}
```
Comment (ezr-ondrej) [inline pkg/clients/imagebuilder/client.go:391]: I think we can inline it and that way not initiate when not needed :)

```suggestion
	if feature.BlueprintCustomInstaller.IsEnabled() {
		req.Customizations.Installer = &CustomInstaller{
			Unattended:   true,
			SudoNoPasswd: []string{"@wheel"},
		}
	}
```

**Review by ezr-ondrej — COMMENTED** [review]
Styling nitpicks, the change make sense but I didn't tested it.

File: pkg/clients/imagebuilder/client.go (hunk 1 above)
Comment (loadtheaccumulator) [reply pkg/clients/imagebuilder/client.go:351]: works for me. coming up...

File: pkg/clients/imagebuilder/client.go (hunk 2 above)
Comment (loadtheaccumulator) [reply pkg/clients/imagebuilder/client.go:391]: works for me. coming up...

**Review by ezr-ondrej — APPROVED** [review]
Cool :+1:
Can we merge, or do we need to synchronize with the removal of the installer hook?

**Pull request merged**